cmd/zpool/zpool_main.c

```diff
@@ -676,6 +676,20 @@ zpool_do_add(int argc, char **argv)
 		return (1);
 	}
 
+	/* unless manually specified use "ashift" pool property for new vdevs */
+	if (!nvlist_exists(props, ZPOOL_CONFIG_ASHIFT)) {
```

Comment:
@loli10K so I completely understand your legitimate concerns regarding making this a changeable property.  

However, I think we should change it to PROP_DEFAULT and clearly explain in the man page that this is a _hint_ for the preferred ashift to use for new devices.  This shouldn't be too confusing since lots of existing properties (compression, dedup, blocksize) already behave in a similar why with the updated value only impacting new files.  More importantly, making this configurable gives us a way to resolve a legitimate issue at some sites.

At this point there are a large number of old existing pools which were created with 512b drives and no explicit ashift set at pool creation time.  This means that when replacing a drive in one of these systems with a 4k spare you must specify the with `-o ashift=<n>`.  That's fine for manual repairs since we want to administrator to know this could negatively impact performance.  However, it's problematic when you're relying on the ZED to automatically kick in a hotspare or do an auto-replace.  In this case, you need `zpool replace` to do a reasonable thing and not fail.  Letting administrators of existing pools set this after the fact handles this problem.

But for this to work you'll also need to add a block similar to this one to `zpool_do_attach_or_replace()`.